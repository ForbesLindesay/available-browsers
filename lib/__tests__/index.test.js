const getCapabilities = require('../');
const {browsers} = require('../');

test('browserstack', () => {
  expect(getCapabilities('browserstack', 'chrome')).toMatchSnapshot();
});

test('saucelabs', () => {
  expect(getCapabilities('saucelabs', 'chrome')).toMatchSnapshot();
});

test('testingbot', () => {
  expect(getCapabilities('testingbot', 'chrome')).toMatchSnapshot();
});

test('browsers', () => {
  const remotes = Object.keys(browsers);
  const browserCount = {};
  const expectedBrowsers = [];
  remotes.forEach(remote => {
    Object.keys(browsers[remote]).forEach(browserName => {
      browserCount[browserName] = browserCount[browserName] || 0;
      browserCount[browserName]++;
      if (
        browserCount[browserName] > 1 &&
        !expectedBrowsers.includes(browserName)
      ) {
        expectedBrowsers.push(browserName);
      }
    });
  });

  remotes.forEach(remote => {
    expect(
      Object.keys(browsers[remote])
        .filter(browserName => {
          return browserCount[browserName] === 1;
        })
        .map(browserName => ({
          type: 'add',
          browserName,
        }))
        .concat(
          expectedBrowsers
            .filter(browserName => !browsers[remote][browserName])
            .map(browserName => ({
              type: 'remove',
              browserName,
            })),
        )
        .sort((a, b) => (a.browserName < b.browserName ? 1 : -1))
        .map(b => (b.type === 'add' ? ' + ' : ' - ') + b.browserName)
        .join('\n'),
    ).toMatchSnapshot(remote + ' browser diff');
  });
});
