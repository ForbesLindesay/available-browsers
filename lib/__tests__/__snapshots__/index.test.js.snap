// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`browserstack 1`] = `
Object {
  "browserName": "chrome",
  "browser_version": "58.0",
  "os": "Windows",
  "os_version": "10",
}
`;

exports[`browserstack browser diff 1`] = `""`;

exports[`saucelabs 1`] = `
Object {
  "browserName": "chrome",
  "platform": "Windows 10",
  "version": "58.0",
}
`;

exports[`saucelabs browser diff 1`] = `""`;

exports[`testingbot 1`] = `
Object {
  "browserName": "chrome",
  "platform": "WIN10",
  "version": "58",
}
`;

exports[`testingbot browser diff 1`] = `
" + phantomjs
 + opera_tablet
 + opera_phone
 + firefox-unbranded"
`;
